test/system/projects/samples_test.rb

```diff
@@ -941,5 +941,52 @@ class SamplesTest < ApplicationSystemTestCase
         assert_selector 'tr:last-child td:first-child', text: @sample3.name
       end
     end
+
+    test 'should not import metadata' do
+      login_as users(:ryan_doe)
+      visit namespace_project_samples_url(@namespace, @project)
+      assert_text I18n.t('projects.samples.index.import_metadata_button'), count: 0
+    end
+
+    test 'should import metadata' do
+      visit namespace_project_samples_url(@namespace, @project)
+      click_link I18n.t('projects.samples.index.import_metadata_button'), match: :first
+      within('span[data-controller-connected="true"] dialog') do
+        attach_file 'file_import[file]', Rails.root.join('test/fixtures/files/metadata/valid.csv')
+        find('#file_import_sample_id_column', wait: 1).find(:xpath, 'option[2]').select_option
+        click_on I18n.t('projects.samples.metadata.file_imports.dialog.submit_button')
+      end
+      within %(turbo-frame[id="import_metadata_dialog"]) do
+        assert_text I18n.t('projects.samples.metadata.file_imports.success.description')
+        click_on I18n.t('projects.samples.metadata.file_imports.success.ok_button')
+      end
+    end
+
+    test 'should import metadata with file validation errors' do
+      visit namespace_project_samples_url(@namespace, @project)
+      click_link I18n.t('projects.samples.index.import_metadata_button'), match: :first
+      within('span[data-controller-connected="true"] dialog') do
+        attach_file 'file_import[file]', Rails.root.join('test/fixtures/files/metadata/missing_metadata_rows.csv')
+        find('#file_import_sample_id_column', wait: 1).find(:xpath, 'option[2]').select_option
+        click_on I18n.t('projects.samples.metadata.file_imports.dialog.submit_button')
+      end
+      within %(turbo-frame[id="import_metadata_dialog_alert"]) do
+        assert_text I18n.t('services.samples.metadata.import_file.missing_metadata_row')
+      end
+    end
+
+    test 'should import metadata with missing sample errors' do
+      visit namespace_project_samples_url(@namespace, @project)
+      click_link I18n.t('projects.samples.index.import_metadata_button'), match: :first
+      within('span[data-controller-connected="true"] dialog') do
+        attach_file 'file_import[file]', Rails.root.join('test/fixtures/files/metadata/mixed_project_samples.csv')
+        find('#file_import_sample_id_column', wait: 1).find(:xpath, 'option[2]').select_option
+        click_on I18n.t('projects.samples.metadata.file_imports.dialog.submit_button')
+      end
+      within %(turbo-frame[id="import_metadata_dialog"]) do
+        assert_text I18n.t('projects.samples.metadata.file_imports.errors.description')
+        click_on I18n.t('projects.samples.metadata.file_imports.errors.ok_button')
+      end
+    end
```

Comment:
Can you please add tests for these following cases. We have them in the service but should also test them in the UI: 

1. 'duplicate headers'
2. `missing header row`
3. missing metadata column(s)
4. user cannot see the import metadata button if they don't have the correct permissions
5. import metadata with an excel file, 
6. import metadata with an invalid file type
7. import metadata with `ignore empty values` set to `true` and `false`

Reply:
aad941f0c7f5201d818990e574fa8db57db76bc2